lib/apis/InSkillPurchase.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const _ = require('lodash');
+const rp = require('request-promise');
+
+class InSkillPurchase {
+  constructor(alexaEvent) {
+    this.alexaEvent = alexaEvent;
+  }
+
+  isAllowed() {
+    const ALLOWED_ISP_ENDPOINTS = {
+      'en-US': 'https://api.amazonalexa.com',
+    };
+
+    const locale = this.alexaEvent.request.locale;
+    const endpoint = _.get(this.alexaEvent, 'context.System.apiEndpoint');
+
+    return ALLOWED_ISP_ENDPOINTS[locale] === endpoint;
+  }
+
+  static buy(productId, token) {
+    return {
+      type: 'Connections.SendRequest',
+      name: 'Buy',
+      payload: {
+        InSkillProduct: {
+          productId,
+        },
+      },
+      token: token || 'token',
+    };
+  }
+
+  static cancel(productId, token) {
+    return {
+      type: 'Connections.SendRequest',
+      name: 'Cancel',
+      payload: {
+        InSkillProduct: {
+          productId,
+        },
+      },
+      token: token || 'token',
+    };
+  }
+
+  static upsell(productId, upsellMessage, token) {
+    return {
+      type: 'Connections.SendRequest',
+      name: 'Upsell',
+      payload: {
+        InSkillProduct: {
+          productId,
+        },
+        upsellMessage,
+      },
+      token: token || 'token',
+    };
+  }
+
+  buyByReferenceName(referenceName, token) {
+    return this.getProductByReferenceName(referenceName)
```

Comment:
await/async ?

Reply:
Travis runs tests on NodeJS 6. I did have them, but I had to take them out because of this